src/server/index.js

```diff
@@ -1,3 +1,8 @@
+// configure env vars in dev and test environments
+if (['development', 'test'].includes(process.env.NODE_ENV)) {
```

Comment:
`if (process.env.NODE_ENV !== 'production') {...}`?

Reply:
I generally prefer to specify the positive case, for the sake of making things overall more deterministic, but I'll come back to this after taking a closer look at the existing problems with this PR.